src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/OsxIPGlobalProperties.cs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace System.Net.NetworkInformation
+{
+    internal class OsxIPGlobalProperties : IPGlobalProperties
+    {
+        public override string DhcpScopeName
+        {
+            get
+            {
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public override string DomainName
+        {
+            get
+            {
+                return HostInformation.DomainName;
+            }
+        }
+
+        public override string HostName
+        {
+            get
+            {
+                return HostInformation.HostName;
+            }
+        }
+
+        public override bool IsWinsProxy
+        {
+            get
+            {
+                return false;
```

Comment:
As previously noted, WINS is supported on *NIX. If the information is uncertain, we should instead throw `PlatformNotSupportedException`.